### ISSUE TITLE
Implemention of sparse version GAT layer

### ISSUE THREAD

**Issue by xiao-kong-long** [AnoushkaVyas/GraphZoo#3]
Dear developer, thanks for your contribution in hyperbolic graph neural network ! Your work makes it convinent to apply hyperbolic space network to analyze data.
In the application of HGAT, I might find a problem in implementation of sparse version GAT layer (SpGraphAttentionLayer). In caculation of edge_h variable, I thought it may correspond to combination of adjcent matrix and attention coefficient in GAT paper, however, the code used square distance function to compute the attention coefficient, which I thought not suitable. Because larger square distance means lower similarity between two nodes, we should assign a smaller value to attention coefficient.
There is another question, in the process of softmax operation, this code first caculated square distance and then made exponential operations, which would produce extremely large number even np.inf. The dataload part of GraphZoo designed nomalization function to solve this problem. But nomalization operation caused very small values in node features especially when dimension is large, which makes parameters learning extremely slow (my loss function value is about 1e-6).
So, if there are better way to caculate attention coefficient? May 1/distence or 1/sqrt(distance) works?
Thanks for your any reply !

**Comment by xiao-kong-long** [comment]
sorry, I have found the basis of attention coefficient fomura in https://ieeexplore.ieee.org/document/9435103
